common/web/keyboard-processor/src/text/stringDivergence.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Returns the index for the code point divergence point in code unit coordinates.
+ * @param str1
+ * @param str2
+ * @param commonRight If false or undefined, asserts a common prefix to the strings.  If true, asserts a common suffix.
```

Comment:
> And instead of `commonRight` we should say `commonSuffix` to avoid confusion with RTL/LTR scripts.

Fair.

> I think it will be better split.

I find myself disagreeing for a few reasons:
- While the "common prefix" version is fairly intuitive... the "common suffix" one is _markedly_ less so, or at least it is to me.  I find that having the merged method _heavily_ clarifies the logic underlying the "common suffix" pathway.
- Any future fixes, like with https://github.com/keymanapp/keyman/pull/10662#discussion_r1480981258 would be subjected to WETness, so it'd easily be possible to miss fixing one version or the other.  For example, the linked comment was directed toward the "prefix" variant, but the "suffix" variant also has an equivalent case.
- Being WET = more filesize cost.  Not _that_ big a footprint here... I think... but there is still a difference.

Reply:
Complexity argument:
* The original function includes 5 separate branches based on that parameter -- including one buried in the test predicate of the `for` loop
* `calculateCommonPrefix` has 2 variables where the original function has 12. `calculateCommonSuffix` should have 3 variables.

Yes, it's a little WETter, but it's far more maintainable and easier to reason against. There are no performance benefits to the DRYer solution -- it will necessarily be slower with the extra tests. The size benefit is marginal, if any.

Reply:
> * `calculateCommonPrefix` has 2 variables where the original function has 12. `calculateCommonSuffix` should have 3 variables.

I now count (effectively) 8.  This does _not_ include:
- commonPotentialSurrogate
- divergentChar1
- divergentChar2

Those three can easily be eliminated by having their assigned values inlined for their single usage within the method.  It's not fair to count them against the original implementation.

- isHigh
- isLow

Those two aren't included, as you didn't count them for the split versions.

I could eliminate more variables for the merged version, but the result would almost certainly be _less_ clear if I did so.  I find breaking things down into pieces with them helps with clarity here.